torchgeo/datasets/mmearth.py

```diff
@@ -0,0 +1,505 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""MM-Earth Dataset."""
+
+import json
+import os
+from collections.abc import Callable, Sequence
+from typing import Any, ClassVar, cast
+
+import numpy as np
+import torch
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, lazy_import
+
+
+class MMEarth(NonGeoDataset):
+    """MM-Earth dataset.
+
+    There are three different versions of the dataset, that vary in image size
+    and the number of tiles:
+
+    * MMEarth: 128x128 px, 1.2 M tiles, 579 GB
+    * MMEarth64: 64x64 px, 1.2 M tiles, 162 GB
+    * MMEarth100k: 128x128 px, 100 K tiles, 48 GB
+
+    The dataset consists of 12 modalities:
+
+    * Aster: elevation and slope
+    * Biome: 14 terrestrial ecosystem categories
+    * ETH Canopy Height: Canopy height and standard deviation
+    * Dynamic World: 9 landcover categories
+    * Ecoregion: 846 ecoregion categories
+    * era5: Climate reanalysis data for temperature mean, min, and max of [year, month, previous month]
```

Comment:
```suggestion
    * ERA5: Climate reanalysis data for temperature mean, min, and max of [year, month, previous month]
```
Since everything else is capitalized